### PR TITLE
MEC013 (distance, areaCircle) - MEC012 - MEC028 phase 1 testing fixes

### PR DESCRIPTION
**Pull request by pastorsx**
Bug Fixes coming from testing:
MEC013
1) Distance endpoint error code rework for query parameters combinations
2) Omitting accuracy in locationInfo for shape different from shape 4, 5, 6 (as per spec)
3) Returning callbackData from subscription in callackData in notification (used to return clientCorrelator)
4) Support count to limit the number of notifications for distance and areCircle subscriptions
5) Non existing UE (or disconnected) not taken into account for distance and areaCircle notifications
6) Dynamic States (number of notifications sent and if already in a circle area) not overriden when receiving a PUT that modifies a subscription 
7) Add/DELETE UE triggers ENTERING/LEAVING area circle notification

MEC028
1) constant creation of new metricStore removed (duplicated code coming from bad merge)

### CODE REVIEW

File: go-apps/meep-loc-serv/server/loc-serv.go [hunk 1]
```diff
@@ -147,6 +149,8 @@ var mutex sync.Mutex
 var gisAppClient *gisClient.APIClient
 var gisAppClientUrl string = "http://meep-gis-engine"
 
+const badRequestErrorCodeString = "400"
```
Comment (dilallkx) [inline go-apps/meep-loc-serv/server/loc-serv.go:152]: Use `http.StatusBadRequest` instead.

File: go-apps/meep-loc-serv/server/loc-serv.go [hunk 2]
```diff
@@ -384,215 +388,235 @@ func checkNotificationDistancePeriodicTrigger() {
...
+					refAddr := pair.addr1
+					monitoredAddr := pair.addr2
+
+					var distParam gisClient.TargetPoint
+					distParam.AssetName = monitoredAddr
+
+					distResp, _, err := gisAppClient.GeospatialDataApi.GetDistanceGeoDataByName(context.TODO(), refAddr, distParam)
+					if err != nil {
+						//getting distance of an element that is not in the DB (not in scenario, not connected) returns error code 400 (bad parameters) in the API. Using that error code to track that request made it to GIS but no good result, so ignore that address (monitored or ref)
+						errCodeStr := strings.Split(err.Error(), " ")
```
Comment (dilallkx) [inline go-apps/meep-loc-serv/server/loc-serv.go:441]: You should retrieve the status code from the http response rather than parsing the error string.

**Review by dilallkx — APPROVED** [review]

File: go-apps/meep-loc-serv/server/loc-serv.go (hunk 1 above)
Comment (pastorsx) [reply go-apps/meep-loc-serv/server/loc-serv.go:152]: fixed

File: go-apps/meep-loc-serv/server/loc-serv.go (hunk 2 above)
Comment (pastorsx) [reply go-apps/meep-loc-serv/server/loc-serv.go:441]: fixed

**Pull request merged**